### PR TITLE
Filter tasks by desired status in ps

### PR DESCRIPTION
**Pull request by PettitWesley**
*Issue #, if available:*
#400 

*Description of changes:*
Added `--desired-status` flag to ps commands.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ecs-cli/modules/cli/compose/entity/entity_helper.go [hunk 1]
```diff
@@ -144,18 +144,28 @@ func createRegisterTaskDefinitionRequest(taskDefinition *ecs.TaskDefinition) *ec
...
 	}
 	return composecontainer.ConvertContainersToInfoSet(containers), nil
 }
 
+func validateDesiredStatus(desiredStatus string) error {
+	if desiredStatus != "" && desiredStatus != ecs.DesiredStatusRunning && desiredStatus != ecs.DesiredStatusStopped {
+		return fmt.Errorf("%s is not a valid value for desired status. Please use %s or %s.", desiredStatus, ecs.DesiredStatusRunning, ecs.DesiredStatusStopped)
+	}
+	return nil
+}
```
Comment (SoManyHs) [inline ecs-cli/modules/cli/compose/entity/entity_helper.go]: This seems fine for our current behavior, though we do seem to technically allowed `DesiredStatusPending` even though it never does anything, which is confusing. If we knew there'd be more future desired statuses I might suggest making this more extensible but it's good for now.

File: ecs-cli/modules/cli/compose/entity/entity_helper.go [hunk 2]
```diff
@@ -144,18 +144,28 @@ func createRegisterTaskDefinitionRequest(taskDefinition *ecs.TaskDefinition) *ec
 
 // Info returns a formatted list of containers (running and stopped) in the current cluster
 // filtered by this project if filterLocal is set to true
-func Info(entity ProjectEntity, filterLocal bool) (project.InfoSet, error) {
-	containers, err := collectContainers(entity, filterLocal)
+func Info(entity ProjectEntity, filterLocal bool, desiredStatus string) (project.InfoSet, error) {
```
Comment (SoManyHs) [inline ecs-cli/modules/cli/compose/entity/entity_helper.go]: Can you also add a test here for the error case on bad status being passed in?

**Comment by PettitWesley** [comment]
Testing:

```
$ ecs-cli ps --cluster TN --region us-east-2
Name                                            State                Ports                     TaskDefinition       Health
37e873f6-37b4-42a7-af47-eac7275c6152/web        RUNNING              10.0.1.28:8080->8080/tcp  Documents:2          UNKNOWN
37e873f6-37b4-42a7-af47-eac7275c6152/lb         RUNNING              10.0.1.28:80->80/tcp      Documents:2          UNKNOWN
37e873f6-37b4-42a7-af47-eac7275c6152/redis      RUNNING                                        Documents:2          UNKNOWN
40bedf31-d707-446e-affc-766eac4cfb85/mysql      RUNNING                                        ecscompose-test2:21  UNKNOWN
40bedf31-d707-446e-affc-766eac4cfb85/wordpress  RUNNING              52.15.92.6:80->80/tcp     ecscompose-test2:21  UNKNOWN
34a47ca4-9987-46e1-a7ca-5c1c5e254679/mysql      STOPPED ExitCode: 0                            cat:1                UNKNOWN
7666b885-5ed5-4ac7-9b9b-99f8f8d908f9/mysql      STOPPED ExitCode: 0                            cat:1                UNKNOWN
86421369-a9c9-4b7b-a92d-914c7954d345/mysql      STOPPED ExitCode: 0                            cat:1                UNKNOWN
```

```
$ ecs-cli ps --cluster TN --region us-east-2 --desired-status CAT
FATA[0000] Error executing 'ps': CAT is not a valid value for desired status. Please use RUNNING or STOPPED. 
amazon-ecs-cli (desired-status) $ ecs-cli ps --cluster TN --region us-east-2 --desired-status RUNNING
Name                                            State    Ports                     TaskDefinition       Health
37e873f6-37b4-42a7-af47-eac7275c6152/web        RUNNING  10.0.1.28:8080->8080/tcp  Documents:2          UNKNOWN
37e873f6-37b4-42a7-af47-eac7275c6152/lb         RUNNING  10.0.1.28:80->80/tcp      Documents:2          UNKNOWN
37e873f6-37b4-42a7-af47-eac7275c6152/redis      RUNNING                            Documents:2          UNKNOWN
40bedf31-d707-446e-affc-766eac4cfb85/mysql      RUNNING                            ecscompose-test2:21  UNKNOWN
40bedf31-d707-446e-affc-766eac4cfb85/wordpress  RUNNING  52.15.92.6:80->80/tcp     ecscompose-test2:21  UNKNOWN
```

```
$ ecs-cli ps --cluster TN --region us-east-2 --desired-status STOPPED
Name                                        State                Ports  TaskDefinition  Health
34a47ca4-9987-46e1-a7ca-5c1c5e254679/mysql  STOPPED ExitCode: 0         cat:1           UNKNOWN
7666b885-5ed5-4ac7-9b9b-99f8f8d908f9/mysql  STOPPED ExitCode: 0         cat:1           UNKNOWN
86421369-a9c9-4b7b-a92d-914c7954d345/mysql  STOPPED ExitCode: 0         cat:1           UNKNOWN
```

File: ecs-cli/modules/cli/compose/entity/entity_helper.go (hunk 2 above)
Comment (SoManyHs) [reply ecs-cli/modules/cli/compose/entity/entity_helper.go]: Looks like `TestTaskInfoFilterLocal` is failing in the tests...

Comment (PettitWesley) [reply ecs-cli/modules/cli/compose/entity/entity_helper.go]: Sure, I added a test in `entity_helper_test.go`

**Pull request merged**